src/cursor.rs

```diff
@@ -57,12 +57,12 @@ pub trait Cursor<'txn> {
     /// For databases with duplicate data items (`DatabaseFlags::DUP_SORT`), the
     /// duplicate data items of each key will be returned before moving on to
     /// the next key.
-    fn iter_from<K>(&mut self, key: K) -> Result<Iter<'txn>> where K: AsRef<[u8]> {
+    fn iter_from<K>(&mut self, key: K) -> Box<Iterator<Item=Result<(&'txn [u8], &'txn [u8])>>> where K: AsRef<[u8]> {
         match self.get(Some(key.as_ref()), None, ffi::MDB_SET_RANGE) {
             Ok(_) | Err(Error::NotFound) => (),
-            Err(error) => return Err(error),
+            Err(error) => return Box::new(iter::once(Err(error))),
```

Comment:
Here, I see how you will only get an error once (and then the iterator ends).  With the other approach (`Item = Result<...>`), how do you ensure you don't get `Err` forever?  This feels similar to `fuse` in spirit.

Reply:
> Here, I see how you will only get an error once (and then the iterator ends). With the other approach (`Item = Result<...>`), how do you ensure you don't get `Err` forever? This feels similar to `fuse` in spirit.

I don't currently ensure that; rather, I leave it to the caller to decide what to do on error. I'm not entirely sure if it's possible for LMDB to return a value after returning an error. If so, then consumers should be able to continue iterating in that case.

Otherwise, consumers that call `Iterator.next()` directly can easily stop iterating at the first error, since they need to match on the return value anyway to distinguish between the `Some(Ok)`, `Some(Err)` and `None` types; and collections can easily automatically stop collecting at the first error; so it seems like consumers wouldn't gain much by the _Iter_ returning `None` after that.